arches_for_science/media/js/utils/xy-parser.js

```diff
@@ -0,0 +1,54 @@
+define([
+], function() {
+
+    const average = (yValues) => {
+        return yValues.reduce((total, num) => total + num, 0) / yValues.length;
+    };
+
+    const runTransformation = (yValues, transform) => {
+        switch(transform) {
+            case 'mean':
+                return average(yValues);
+            default:
+                return yValues[0];
+        }
+    };
+
+    return {
+        transformations: () => {
+            return ["mean"]
+        },
+        parse: (text, config) => {
+            let values;
+            let workingText = text;
+            const parsedData = {x: [], y: []};
+            try {
+                if(config?.footerDelimiter){
+                    workingText = workingText.split(config?.footerDelimiter)[0].trim();
+                }
+                if(config?.headerDelimiter){
+                    values = workingText.split(config?.headerDelimiter)[1].trim().split('\n');
+                } else if (config?.headerFixedLines) {
+                    const lines = workingText.split('\n');
+                    values = lines.slice(config?.headerFixedLines);
+                } else {
+                    values = workingText.trim().split('\n'); 
+                }
+            } catch(e) {
+                values = workingText.trim().split('\n');
+            }
+            const delimiterCharacter = config?.delimiterCharacter ?? ',';
+
+            const valueRegex = (delimiterCharacter.length < 2) ? new RegExp(`[${delimiterCharacter}\\s]+`) : new RegExp(`${delimiterCharacter}`);
```

Comment:
This regex silently failed when I used `**` as a delimiter. This is a pretty unlikely delimiter, but it would be nice to have a message on the front end stating that the delimiter cannot be parsed - just in case.